kernel/kernel-default/src/main/java/io/delta/kernel/parquet/ParquetConverters.java

```diff
@@ -530,6 +530,29 @@ public void resizeIfNeeded()
         }
     }
 
+    public static class FileRowIndexColumnConverter
+            extends LongColumnConverter {
+
+        public FileRowIndexColumnConverter(int initialBatchSize) {
+            super(initialBatchSize);
+        }
+
+        @Override
+        public void addLong(long value) {
+            throw new UnsupportedOperationException("cannot add long to metadata column");
+        }
+
+        /**
+         * @param fileRowIndex the file row index of the row processed
+         */
+        // If moveToNextRow() is called instead the value will be null
+        public boolean moveToNextRow(long fileRowIndex) {
```

Comment:
Wondering if we change the existing `moveToNextRow()` to `moveToNextRow(long index)` simplifies?

I am ok if we want to address this in followup PRs.

Reply:
Yeah I think this would simplify it. And removes the risk of calling the other API on a FileRowIndexColumnConverter.